{"0": {"author": "Duckling", "date": "1685114434398", "content": "Hello! If applicable, put your complete code example down below. Make sure that your code:\n\nis 100% self-contained \u2014 someone can copy-paste exactly what is here and run it to\nreproduce the behaviour you are observing\nincludes comments\n\n# Put code here\n```backend = FakeAlmadenV2()\nnoise_model = NoiseModel.from_backend(backend)\ndev = qml.device(\"qiskit.aer\", wires=n_qubits, noise_model=noise_model)\n\ndef qnode(inputs, w000, w001, w002, w003,\n          w016, w017, w018, w019,\n          w032, w033, w034, w035,\n          w100, w101, w102, w103,\n          w116, w117, w118, w119,\n          w132, w133, w134, w135):\n    qml.AmplitudeEmbedding(features=inputs, wires=range(4), normalize=True)\n    qml.RX(w000, wires=0)\n    qml.RX(w001, wires=1)\n    qml.RX(w002, wires=2)\n    qml.RX(w003, wires=3)\n\n    qml.RZ(w016, wires=0)\n    qml.RZ(w017, wires=1)\n    qml.RZ(w018, wires=2)\n    qml.RZ(w019, wires=3)\n\n    qml.CNOT(wires=[0, 1])\n    qml.CNOT(wires=[1, 2])\n    qml.CNOT(wires=[2, 3])\n    qml.CNOT(wires=[3, 0])\n\n    return (qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1)), qml.expval(qml.PauliZ(2)), qml.expval(qml.PauliZ(3)))\n\n\nIf you want help with diagnosing an error, please put the full error message below:\n\n\n\nPut full error message here\nqiskit.exceptions.QiskitError: \u2018Sum of amplitudes-squared does not equal one.\u2019\nAnd, finally, make sure to include the versions of your packages. Specifically, show us the output of `qml.about()`. \nPlatform info:           Linux-5.15.0-72-generic-x86_64-with-debian-bullseye-sid\nPython version:          3.7.0\nNumpy version:           1.21.6\nScipy version:           1.7.3\nInstalled devices:\n- lightning.gpu (PennyLane-Lightning-GPU-0.26.0)\n- lightning.qubit (PennyLane-Lightning-0.26.0)\n- default.gaussian (PennyLane-0.26.0)\n- default.mixed (PennyLane-0.26.0)\n- default.qubit (PennyLane-0.26.0)\n- default.qubit.autograd (PennyLane-0.26.0)\n- default.qubit.jax (PennyLane-0.26.0)\n- default.qubit.tf (PennyLane-0.26.0)\n- default.qubit.torch (PennyLane-0.26.0)\n- default.qutrit (PennyLane-0.26.0)\n- qiskit.aer (PennyLane-qiskit-0.24.0)\n- qiskit.basicaer (PennyLane-qiskit-0.24.0)\n- qiskit.ibmq (PennyLane-qiskit-0.24.0)\n- qiskit.ibmq.circuit_runner (PennyLane-qiskit-0.24.0)\n- qiskit.ibmq.sampler (PennyLane-qiskit-0.24.0)\n- ionq.qpu (PennyLane-IonQ-0.23.0)\n- ionq.simulator (PennyLane-IonQ-0.23.0)\n- projectq.classical (PennyLane-PQ-0.17.0)\n- projectq.ibm (PennyLane-PQ-0.17.0)\n- projectq.simulator (PennyLane-PQ-0.17.0)\nNone", "link": "https://discuss.pennylane.ai//t/the-use-of-fake-backend-will-make-amplitude-encoding-unusable/2978/1"}, "1": {"author": "isaacdevlugt", "date": "1685115070369", "content": "Hey @Duckling! Welcome to the forum \nCan you please provide a full example that gives you the error you\u2019re seeing? Right now there\u2019s just a QNode and nothing being executed. Make sure to copy-paste exactly what you\u2019re running on your machine so that I can try to replicate it 1 Reply", "link": "https://discuss.pennylane.ai//t/the-use-of-fake-backend-will-make-amplitude-encoding-unusable/2978/2"}, "2": {"author": "Duckling", "date": "1685115234567", "content": "Traceback (most recent call last):\nFile \u201c/mnt/data2/wrh/QSteal/Four class/Noisy_Cloud_Ampli_Circuit2.py\u201d, line 211, in \ntest(model, device, test_loader)\nFile \u201c/mnt/data2/wrh/QSteal/Four class/Noisy_Cloud_Ampli_Circuit2.py\u201d, line 193, in test\noutputs = model(images)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/torch/nn/modules/module.py\u201d, line 1130, in _call_impl\nreturn forward_call(*input, **kwargs)\nFile \u201c/mnt/data2/wrh/QSteal/Four class/Noisy_Cloud_Ampli_Circuit2.py\u201d, line 151, in forward\nx = self.qlayer(x)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/torch/nn/modules/module.py\u201d, line 1130, in _call_impl\nreturn forward_call(*input, **kwargs)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/qnn/torch.py\u201d, line 309, in forward\nreconstructor = [self.forward(x) for x in torch.unbind(inputs)]\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/qnn/torch.py\u201d, line 309, in \nreconstructor = [self.forward(x) for x in torch.unbind(inputs)]\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/qnn/torch.py\u201d, line 313, in forward\nreturn self._evaluate_qnode(inputs)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/qnn/torch.py\u201d, line 328, in _evaluate_qnode\nreturn self.qnode(**kwargs).type(x.dtype)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/qnode.py\u201d, line 668, in call\n**self.execute_kwargs,\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/interfaces/execution.py\u201d, line 444, in execute\ntapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_diff=max_diff, mode=_mode\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/interfaces/torch.py\u201d, line 248, in execute\nreturn ExecuteTapes.apply(kwargs, *parameters)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/interfaces/torch.py\u201d, line 85, in forward\nres, ctx.jacs = ctx.execute_fn(ctx.tapes, **ctx.gradient_kwargs)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/interfaces/execution.py\u201d, line 197, in wrapper\nres = fn(execution_tapes.values(), **kwargs)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/interfaces/execution.py\u201d, line 122, in fn\nreturn original_fn(tapes, **kwargs)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/contextlib.py\u201d, line 74, in inner\nreturn func(*args, **kwds)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane_qiskit/qiskit_device.py\u201d, line 444, in batch_execute\ncompiled_circuits = self.compile_circuits(circuits)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane_qiskit/qiskit_device.py\u201d, line 433, in compile_circuits\nself.create_circuit_object(circuit.operations, rotations=circuit.diagonalizing_gates)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane_qiskit/qiskit_device.py\u201d, line 246, in create_circuit_object\napplied_operations = self.apply_operations(operations)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane_qiskit/qiskit_device.py\u201d, line 307, in apply_operations\ngate = mapped_operation(*par)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/qiskit/extensions/quantum_initializer/initializer.py\u201d, line 57, in init\nself._stateprep = StatePreparation(params, num_qubits)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/qiskit/circuit/library/data_preparation/state_preparation.py\u201d, line 99, in init\nnum_qubits = self._get_num_qubits(num_qubits, params)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/qiskit/circuit/library/data_preparation/state_preparation.py\u201d, line 202, in _get_num_qubits\nraise QiskitError(\u201cSum of amplitudes-squared does not equal one.\u201d)\nqiskit.exceptions.QiskitError: \u2018Sum of amplitudes-squared does not equal one.\u2019", "link": "https://discuss.pennylane.ai//t/the-use-of-fake-backend-will-make-amplitude-encoding-unusable/2978/3"}, "3": {"author": "Duckling", "date": "1685115492354", "content": "Hello, it\u2019s my first time to use the support. Some operations are not very familiar. \nThe following is my error output:\nTraceback (most recent call last):\nFile \u201c/mnt/data2/wrh/QSteal/Four class/Noisy_Cloud_Ampli_Circuit2.py\u201d, line 211, in \ntest(model, device, test_loader)\nFile \u201c/mnt/data2/wrh/QSteal/Four class/Noisy_Cloud_Ampli_Circuit2.py\u201d, line 193, in test\noutputs = model(images)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/torch/nn/modules/module.py\u201d, line 1130, in _call_impl\nreturn forward_call(*input, **kwargs)\nFile \u201c/mnt/data2/wrh/QSteal/Four class/Noisy_Cloud_Ampli_Circuit2.py\u201d, line 151, in forward\nx = self.qlayer(x)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/torch/nn/modules/module.py\u201d, line 1130, in _call_impl\nreturn forward_call(*input, **kwargs)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/qnn/torch.py\u201d, line 309, in forward\nreconstructor = [self.forward(x) for x in torch.unbind(inputs)]\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/qnn/torch.py\u201d, line 309, in \nreconstructor = [self.forward(x) for x in torch.unbind(inputs)]\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/qnn/torch.py\u201d, line 313, in forward\nreturn self._evaluate_qnode(inputs)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/qnn/torch.py\u201d, line 328, in _evaluate_qnode\nreturn self.qnode(**kwargs).type(x.dtype)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/qnode.py\u201d, line 668, in call\n**self.execute_kwargs,\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/interfaces/execution.py\u201d, line 444, in execute\ntapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_diff=max_diff, mode=_mode\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/interfaces/torch.py\u201d, line 248, in execute\nreturn ExecuteTapes.apply(kwargs, *parameters)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/interfaces/torch.py\u201d, line 85, in forward\nres, ctx.jacs = ctx.execute_fn(ctx.tapes, **ctx.gradient_kwargs)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/interfaces/execution.py\u201d, line 197, in wrapper\nres = fn(execution_tapes.values(), **kwargs)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane/interfaces/execution.py\u201d, line 122, in fn\nreturn original_fn(tapes, **kwargs)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/contextlib.py\u201d, line 74, in inner\nreturn func(*args, **kwds)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane_qiskit/qiskit_device.py\u201d, line 444, in batch_execute\ncompiled_circuits = self.compile_circuits(circuits)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane_qiskit/qiskit_device.py\u201d, line 433, in compile_circuits\nself.create_circuit_object(circuit.operations, rotations=circuit.diagonalizing_gates)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane_qiskit/qiskit_device.py\u201d, line 246, in create_circuit_object\napplied_operations = self.apply_operations(operations)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/pennylane_qiskit/qiskit_device.py\u201d, line 307, in apply_operations\ngate = mapped_operation(*par)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/qiskit/extensions/quantum_initializer/initializer.py\u201d, line 57, in init\nself._stateprep = StatePreparation(params, num_qubits)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/qiskit/circuit/library/data_preparation/state_preparation.py\u201d, line 99, in init\nnum_qubits = self._get_num_qubits(num_qubits, params)\nFile \u201c/home/ruhan/anaconda3/envs/Pytorch/lib/python3.7/site-packages/qiskit/circuit/library/data_preparation/state_preparation.py\u201d, line 202, in _get_num_qubits\nraise QiskitError(\u201cSum of amplitudes-squared does not equal one.\u201d)\nqiskit.exceptions.QiskitError: \u2018Sum of amplitudes-squared does not equal one.\u2019", "link": "https://discuss.pennylane.ai//t/the-use-of-fake-backend-will-make-amplitude-encoding-unusable/2978/4"}, "4": {"author": "isaacdevlugt", "date": "1685115564374", "content": "Can you share your code that you\u2019re running?", "link": "https://discuss.pennylane.ai//t/the-use-of-fake-backend-will-make-amplitude-encoding-unusable/2978/5"}, "5": {"author": "isaacdevlugt", "date": "1685368368340", "content": "Hey @Duckling! I see that you deleted your message. Were you able to solve your issue? If not, what I was able to copy-paste before you deleted your code seems to be missing test_loader.", "link": "https://discuss.pennylane.ai//t/the-use-of-fake-backend-will-make-amplitude-encoding-unusable/2978/6"}, "6": {"author": "isaacdevlugt", "date": "1685369823348", "content": "Also, this is probably unrelated to the error you\u2019re getting, but you should consider updating to v0.30 of Pennylane! 1", "link": "https://discuss.pennylane.ai//t/the-use-of-fake-backend-will-make-amplitude-encoding-unusable/2978/7"}}